src/apps/pillarx-app/index.tsx

```diff
@@ -1,39 +1,75 @@
+import { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import { useTranslation } from 'react-i18next';
 
+// types
+import { Projection } from '../../types/api';
+
 // hooks
-import { useApiData } from './hooks/useApiData';
+import { useGetTilesInfoQuery } from './api/apiSlice';
+import { useWalletAddress } from '@etherspot/transaction-kit';
 
 // components
 import PillarXLogo from './components/PillarXLogo/PillarXLogo';
 import pillarLogoLight from './images/pillarX_full_white.png';
 import H1 from './components/Typography/H1';
-import PortfolioOverview from './components/PortfolioOverview/PortfolioOverview';
-import TokensHorizontalTile from './components/TokensHorizontalTile/TokensHorizontalTile';
-import TokensVerticalTile from './components/TokensVerticalTile/TokensVerticalTile';
+import { componentMap } from './utils/configComponent';
+import SkeletonTiles from './components/SkeletonTile/SkeletonTile';
 
 export const App = () => {
   const [t] = useTranslation();
+  const [page, setPage] = useState(1);
+  const [pageData, setPageData] = useState<Projection[]>([]);
+
+  const walletAddress = useWalletAddress();
+  const { data: apiData, isLoading: isApiLoading, isFetching } = useGetTilesInfoQuery( { page: page, address: walletAddress || '' });
+
+  useEffect(() => {
+    // when apiData loads, we save it in a state to keep previous data
+      if (apiData) {
+          setPageData((prevData) => [...prevData, ...apiData.projection]);
+      }
+  }, [apiData]);
+
+  // scroll handler makes sure that when reaching the end of the page, it loads the next page
+  const handleScroll = () => {
+      if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
+          setPage((prevPage) => prevPage + 1);
+      }
+  };
 
-  const {
-    isLoading,
-    dataPortlioOverview,
-    dataTokensHorizontal,
-    dataTokensVerticalLeft,
-    dataTokensVerticalRight,
-    titleTokensHorizontal,
-    titledataTokensVerticalLeft,
-    titledataTokensVerticalRight,
-  } = useApiData();
+  useEffect(() => {
+      window.addEventListener('scroll', handleScroll);
+      return () => window.removeEventListener('scroll', handleScroll);
+  }, []);
+
+
+
+const displayAllTiles = () => {
+  const allTileComponents = [];
+  
+  for (let index = 0; index < pageData.length; index++) {
+    const tileData = pageData[index];
+    
+    const TileComponent = componentMap[tileData.layout];
+    
+    if (TileComponent) {
+      allTileComponents.push(<TileComponent key={index} data={tileData} isDataLoading={isApiLoading} />);
+    } else {
+      return null;
+    }
```

Comment:
I don't think you need to return `null` here? The `for` loop doesn't return anything, it's just an iterator

Reply:
Ah yes sorry